packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -789,25 +789,37 @@ function getPathParameters(operation: Operation) {
   for (const param of operation.parameters) {
     if (param.location === "path") {
       if (!param.optional) {
-        pathParams += `${pathParams !== "" ? "," : ""} ${param.clientName}`;
+        pathParams += `${pathParams !== "" ? "," : ""} ${getParamExpression(
+          param
+        )}`;
         continue;
       }
 
       const defaultValue = getDefaultValue(param);
 
-      pathParams += `${pathParams !== "" ? "," : ""} options.${
-        param.clientName
-      }`;
+      pathParams += `${pathParams !== "" ? "," : ""} ${getParamExpression(
+        param
+      )}`;
 
       if (defaultValue) {
-        pathParams += ` ?? "${defaultValue}"`;
+        pathParams += ` ?? "${getParamExpression(param, defaultValue)}"`;
       }
     }
   }
 
   return pathParams;
 }
 
+function getParamExpression(param: Parameter, defaultValue?: string) {
+  const value = defaultValue
+    ? `"${defaultValue}"`
```

Comment:
Is it possible default value could be other types instead of string in the url template story?

Reply:
> Have we bumped core-client-rest version in the codegen side?
Yes, it is in pr https://github.com/Azure/autorest.typescript/pull/2851.

Reply:
what is our current logic of handling constant type for path parameter?

Reply:
if path parameter is optional we could generate sth like `options.param ?? "defaultValue"`.

Reply:
Will create an issue to track this.

Reply:
Issue created: https://github.com/Azure/typespec-azure/issues/1728